packages/web-app-files/src/views/Personal.vue

```diff
@@ -198,16 +204,73 @@ export default {
   },
 
   methods: {
-    ...mapActions('Files', ['loadIndicators', 'loadPreview']),
+    ...mapActions('Files', ['loadIndicators', 'loadPreview'], 'showMessage'),
     ...mapMutations('Files', [
       'SELECT_RESOURCES',
       'SET_CURRENT_FOLDER',
       'LOAD_FILES',
       'CLEAR_CURRENT_FILES_LIST',
-      'UPDATE_CURRENT_PAGE'
+      'REMOVE_FILE_FROM_SEARCHED'
     ]),
     ...mapMutations(['SET_QUOTA']),
 
+    async fileDropped(fileIdTarget) {
+      const selected = this.selectedFiles
+      const targetInfo = this.activeFiles.find(e => e.id === fileIdTarget)
+      const targetSelected = selected.some(e => e.id === fileIdTarget)
+      if (targetSelected) return
+      if (targetInfo.type !== 'folder') return
+
+      this.loading = true
+      const targetPath = targetInfo.path + '/'
+      const errors = []
+      const itemsInTarget = await this.getFolderItems(targetPath)
+      for (let i = 0; i < selected.length; i++) {
+        const current = selected[i]
+        const exists = itemsInTarget.some(e => basename(e.name) === current.name)
+        if (exists) {
+          const message = this.$gettext('Resource with name %{name} already exists')
+          errors.push({
+            resource: current.name,
+            message: this.$gettextInterpolate(message, { name: current.name }, true)
+          })
+          continue
+        }
+        try {
+          await this.$client.files.move(current.path, targetPath + current.name)
+          this.REMOVE_FILE_FROM_SEARCHED(current)
+        } catch (error) {
+          error.resource = current.name
+          errors.push(error)
+        }
+      }
+      await this.loadResources(true)
```

Comment:
I don't like the brute-force `loadResources` here. You are already removing the file from the search results on successful move. You could add another mutation to the store for removing it from the `files` state as well (both have to happen!). This way you'd avoid reloading the entire folder. Which could easily be something with 5000 items.